infra/cifuzz/run_fuzzers.py

```diff
@@ -27,61 +27,175 @@
+    logging.info('Fuzz targets: %s', self.fuzz_target_paths)
+    if not self.fuzz_target_paths:
+      logging.error('No fuzz targets were found in out directory: %s.',
+                    self.out_dir)
+      return False
+
+    return True
+
+  def run_fuzz_target(self, fuzz_target_obj):  # pylint: disable=no-self-use
+    """Fuzzes with |fuzz_target_obj| and returns the result."""
+    # TODO(metzman): Make children implement this so that the batch runner can
+    # do things differently.
+    return fuzz_target_obj.fuzz()
+
+  @property
+  def quit_on_bug_found(self):
+    """Property that is checked to determine if fuzzing should quit after first
+    bug is found."""
+    raise NotImplementedError('Child class must implement method')
+
+  def get_fuzz_target_artifact(self, target, artifact_name):
+    """Returns the path of a fuzzing |artifact| named |artifact_name| for
+    |target|."""
+    artifact_name = target.target_name + '-' + artifact_name
+    return os.path.join(self.artifacts_dir, artifact_name)
+
+  def create_fuzz_target_obj(self, target_path, run_seconds):
+    """Returns a fuzz target object."""
+    return fuzz_target.FuzzTarget(target_path,
+                                  run_seconds,
+                                  self.out_dir,
+                                  self.config.project_name,
+                                  sanitizer=self.config.sanitizer)
+
+  def run_fuzz_targets(self):
+    """Runs fuzz targets. Returns True if a bug was found."""
+    fuzzers_left_to_run = len(self.fuzz_target_paths)
+
+    # Make a copy since we will mutate it.
+    fuzz_seconds = self.config.fuzz_seconds
+
+    min_seconds_per_fuzzer = fuzz_seconds // fuzzers_left_to_run
+    bug_found = False
+    for target_path in self.fuzz_target_paths:
+      # By doing this, we can ensure that every fuzz target runs for at least
+      # min_seconds_per_fuzzer, but that other fuzzers will have longer to run
+      # if one ends early.
+      run_seconds = max(fuzz_seconds // fuzzers_left_to_run,
+                        min_seconds_per_fuzzer)
+
+      target = self.create_fuzz_target_obj(target_path, run_seconds)
+      start_time = time.time()
+      testcase, stacktrace = self.run_fuzz_target(target)
+
+      # It's OK if this goes negative since we take max when determining
+      # run_seconds.
+      fuzz_seconds -= time.time() - start_time
+
+      fuzzers_left_to_run -= 1
+      if not testcase or not stacktrace:
+        logging.info('Fuzzer %s finished running without crashes.',
+                     target.target_name)
+        continue
+
+      # We found a bug in the fuzz target.
+      utils.binary_print(b'Fuzzer: %s. Detected bug:\n%s' %
+                         (target.target_name.encode(), stacktrace))
+
+      # TODO(metzman): Do this with filestore.
+      testcase_artifact = self.get_fuzz_target_artifact(target, 'testcase')
+      shutil.move(testcase, testcase_artifact)
```

Comment:
nit: s/testcase/testcase_path/ and s/testcase_artifact/testcase_artifact_path/ to make things clearer. I had to check the relevant initialisations to see what this means.

Same with other paths in this function (e.g. bug_summary_artifact).